woodwork/table_accessor.py

```diff
@@ -1167,14 +1167,14 @@ def _check_index(dataframe, index):
         raise ColumnNotPresentError(
             f"Specified index column `{index}` not found in dataframe"
         )
-    if (
-        index is not None
-        and isinstance(dataframe, pd.DataFrame)
-        and not dataframe[index].is_unique
-    ):
-        # User specifies an index that is in the dataframe but not unique
-        # Does not check for Dask as Dask does not support is_unique
-        raise IndexError("Index column must be unique")
+    if index is not None and isinstance(dataframe, pd.DataFrame):
+        if not dataframe[index].is_unique:
+            # User specifies an index that is in the dataframe but not unique
+            # Does not check for Dask as Dask does not support is_unique
+            raise IndexError("Index column must be unique")
+
+        if dataframe[index].notnull().any():
```

Comment:
Yeah, either `.isnull().any()` or `.notnull().all()` should work but haven't compared performance

Reply:
nvm, disregard

Reply:
Seems like `.isnull().any()` could be faster by a bit because the `.any()` check could stop as soon as it finds a single null value, while the `.all()` check would have to go through all the rows every time to make sure everything is not null. Probably worth confirming though.

Reply:
Yeah I think `.isnull().any()` makes more sense